### PR TITLE
catalog: remove group ancestors and descendants

### LINKED ISSUE #3049: Remove requirements for Group model
The current group model according to the [docs](https://backstage.io/docs/features/software-catalog/descriptor-format) requires ancestors, children and descendant. 

But this means that if my org is:
root => Engineering -> Platform -> Devtools -> TeamA 
I need to write any of these names in 3 places. Now I want to change "Devtools" to "Internal tools" and need to do modifications in multiple places.

If you want to build a tree of the org, the only required element should be the parent, which interestingly enough is the only optional element currently

### PR DESCRIPTION
**Pull request by freben**
Closes #3049

### CODE REVIEW

File: plugins/catalog-backend/src/ingestion/processors/util/org.ts
Comment (freben) [inline plugins/catalog-backend/src/ingestion/processors/util/org.ts:57]: The documentation states that transitive membership should NOT be specified - so it would seem that this code should not be needed. But I chose to keep this behaviour for now (same functionality, but using the new ancestor-less data) instead of changing that concern as well.

Some background:

- We want the emitted _relations_ of user membership in groups to be transitive in the end.
- But we don't want people to HAVE to fill in all of the transitive memberOf values _in the spec_.
- But the rub is, we need to have a global view of all groups, to be able to know the org hierarchy - because groups do not either reference their ancestors and descendants after this PR is merged.
- So it won't be possible to write a post-processor that looks at one user at a time, and emits its transitive membership relations.
- That means that the _reader processor_ probably will be the one that emits those relations, because it's the one that knows of all of the groups at once before emitting them.

**Review by Fox32 — APPROVED** [review]
Nooooo I invested so much time into implementing them in the Microsoft Graph processor 🤣 

But I saw that coming 😉

File: plugins/catalog-backend/src/ingestion/processors/util/org.ts
Comment (Fox32) [reply plugins/catalog-backend/src/ingestion/processors/util/org.ts:57]: Yeah even though I first wanted to skip the transitive ones, your points are exactly the reasons why I ended up with that behavior in the end.

Comment (freben) [reply plugins/catalog-backend/src/ingestion/processors/util/org.ts:57]: Mm. So we'll be in a bit of an intermediate state where we have to figure out where to go with this.

One solution could of course be to continue to have org reader processors call this helper method to actually put the full list in the spec.

- Benefit: To those processors, there's no harm done, it's just one call away to fill in the entire model.
- Benefit: We can make a core post-processor that automatically emits relations from this filled-in field, instead of making all reader processors emit this relation.
- Benefit: We just have to tweak the docs slightly.
- Drawback: There won't be a distinction between direct and indirect memberships, so a consumer that might be interested in seeing the direct ones and render them won't be able to. But this isn't a usage that we're targeting either way. If we want to do that, we may have to rethink the model a bit.
- Drawback: Other use cases like reading hand-written files on disk will not get the transitives filled in, which means that transitive ownership won't work for those cases. But maybe they are rare particularly in production.

**Comment by freben** [comment]
@Rugvip @benjdlambert This one will be interesting - this will be the first test of the stability index on catalog kinds. We assigned a level of 2 to the catalog-model, so we cannot merge this in its current state. I added the do-not-merge label for now.

**Comment by freben** [comment]
Correct me if I'm wrong, but this particular change (deletion of fields) chould be a fairly easy one.

The simplest way:

- In a different PR, mark the two fields as deprecated (in code and in docs). Leave everything else as is. There is no requirement to have the new "api" (an updated type) in place during the deprecation period.
- After two weeks merge this PR, outright deleting the fields, and do NOT update the apiVersion. This is a bit of a shortcut, but since the reader side will continue to work and just disregard the old fields if they are found in yaml data - things keep working.

The drawback then is if the consuming side wants to be able to discern between the old data format and the new one, even storing them side by side in the catalog. If we want to be thorough, something more must be done.

- In a different PR, mark the two fields as deprecated (in code and in docs). Leave everything else as is. There is no requirement to have the new "api" (an updated type) in place during the deprecation period.
- After two weeks, introduce a new apiVersion of the kind side by side with the old one. Here we will have a choice of whether to do conversion from the old to the new in the read pipe - or to keep them both around. This will certainly increase complexity in code.

If we want to introduce the new apiVersion immediately so both live side by side from day one for two weeks, then it's going to be even worse.

@Rugvip @Fox32 @benjdlambert opinions?

**Comment by freben** [comment]
Deprecation happened in https://github.com/backstage/backstage/pull/3392

**Comment by Fox32** [comment]
Should we add a changeset?

**Comment by freben** [comment]
@Fox32 Oh my, for sure - thanks for calling it out

**Comment by Fox32** [comment]
Looks like this could be merged now?

**Comment by freben** [comment]
Yepyep, it's on my list for the morning!

**Review by Rugvip — APPROVED** [review]

**Pull request merged**
